crawler.rb

```diff
@@ -1,39 +1,63 @@
+require 'pry'
+require 'set'
+require 'JSON'
 require 'nokogiri'
 require 'open-uri'
-require 'JSON'
-require 'pry'
 require_relative './document'
-require_relative './document_collection'
 require_relative './url_fetcher'
 
 class Crawler
-	def initialize(document_collection, url_collection)
-		@document_collection = document_collection
-		@url_collection = url_collection
-	end
-
-	def done_crawling?
-		@url_collection.empty?
-	end
-
-	def crawl_next_url
-		url = @url_collection.next_url
-		content = UrlFetcher.fetch(url)
-		document = Document.new(url, content)
-		@document_collection.add_document(document)
-		puts document.domain_hrefs.take(5)
-		document.domain_hrefs.each do |href|
-			@url_collection.add_url(href)
-		end
-	end
-
-	def run(times_to_run)
-		times_to_run.times do
-			if done_crawling? 
-				puts "Crawling complete!"
-				exit 0
-			end
-			crawl_next_url
-		end
-	end
+  attr_accessor :url, :href, :db
+  def initialize(db)
+    @database = db 
+  end
+
+  def done_crawling?
+    @database.execute( "SELECT count(*) FROM URLS_to_crawl WHERE state = 'uncrawled'" )[0][0] == 0
```

Comment:
Also, after this PR if it's not already on your roadmap, you should look into using rails models with migrations. I think this week you're working on learning SQL so this makes sense as-is.